src/br/com/zup/nossacasacodigo/cart/Cart.java

```diff
@@ -0,0 +1,48 @@
+package br.com.zup.nossacasacodigo.cart;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import br.com.zup.nossacasacodigo.book.Book;
+
+public class Cart {
+	private Set<CartItem> items = new HashSet<>();
+	private static BigDecimal finalValue = BigDecimal.ZERO;
+
+	public void addToCart(Book book, int quantity) {
+		if (quantity < 0 || book == null) {
+			throw new IllegalArgumentException("Livro ou quantidade inv�lida");
+		}
+		Optional<CartItem> alreadyInCart = checkIfIsInCart(book);
+		CartItem item;
+		if (alreadyInCart.isPresent()) {
+			item = alreadyInCart.get();
+			finalValue = finalValue.subtract(item.calculateSubtotal());
```

Comment:
Nesse ponto, você está fazendo uma conta para garantir que o valor final fique certo, será que vale a pena um comentário explicando o porquê?

Reply:
nesse caso varia sentido já que temos que fazer essa correção do calculo

Reply:
Oi Larissa, você manter a consistência do valor entre as múltiplas operações é um código basicamente de tracking e que aumenta a chance de bug, já que é mais código que está sendo escrito :). 

Minha sugestão aqui é sempre calcular o total quando o método de total for invocado.